indigo/indigo/src/main/scala/indigo/gameengine/GameLoop.scala

```diff
@@ -43,16 +43,22 @@ final class GameLoop[StartUpData, GameModel, ViewModel](
     _runningTimeReference = time
     val timeDelta: Long = time - lastUpdateTime
 
-    if timeDelta >= gameConfig.frameRateDeltaMillis.toLong - 1 then
-      runFrame(time, timeDelta)
-      gameEngine.platform.tick(gameEngine.gameLoop(time))
-    else gameEngine.platform.tick(loop(lastUpdateTime))
+    gameConfig.frameRateLimit match
```

Comment:
Similar to the comment above, a curried function branching upfront?

Reply:
So I didn't need a curried function since the values are totally fixed, but deciding the branches once ahead of time is a very sneaky idea and it working nicely - thanks!

```scala
  private val runner: (Long, Long, Long) => Unit =
    gameConfig.frameRateLimit match
      case None =>
        (time, timeDelta, lastUpdateTime) =>
          runFrame(time, timeDelta)
          gameEngine.platform.tick(gameEngine.gameLoop(time))

      case Some(fps) =>
        (time, timeDelta, lastUpdateTime) =>
          if timeDelta >= gameConfig.frameRateDeltaMillis.toLong - 1 then
            runFrame(time, timeDelta)
            gameEngine.platform.tick(gameEngine.gameLoop(time))
          else gameEngine.platform.tick(loop(lastUpdateTime))

  def loop(lastUpdateTime: Long): Long => Unit = { time =>
    _runningTimeReference = time
    val timeDelta: Long = time - lastUpdateTime
    runner(time, timeDelta, lastUpdateTime)
  }
```